app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java

```diff
@@ -330,7 +335,84 @@ private String TryResolveName(String baseURI) {
             return baseURI;
         }
     }
-    
+
+    // return value tells if new data was found
+    private boolean downloadTreatments(final NightscoutService nightscoutService, final String hashedSecret, String baseURI, URI uri) throws Exception{
```

Comment:
`downloadTreatments` and `downloadDeviceStatus` share many identical code lines. Please refactor it to avoid these duplications.